src/cartridge/src/cartridge.cpp

```diff
@@ -31,6 +33,17 @@ Cartridge::Cartridge(std::string rom_path) {
 		rom_file.read(&byte_input[0], rom_file_size);
 		data = std::vector<uint8_t>(byte_input.begin(), byte_input.end());
 
+		/**
+		 * Check the validity of the Logo in the BOOT ROM
+		 */
+		meta_data->check_logo_validity(data);
```

Comment:
Instead of calling this here, you can move this to the constructor of `meta_data` and let it call this method on construction. You can also just directly pass the `data` variable to constructor and let it populate all the data. We can still populate all the data even if the logo verification failed.